src/api/participantsRouter.ts

```diff
@@ -111,3 +122,76 @@ participantsRouter.put(
     }
   }
 );
+
+participantsRouter.get(
+  '/:participantId/sharingPermission',
+  hasParticipantAccess,
+  async (req: ParticipantRequest, res: Response) => {
+    const { participant } = req;
+    if (!participant?.siteId) {
+      return res.status(400).send('Site id is not set');
+    }
+    const sharingParticipants = await getSharingParticipants(participant.siteId);
+    return res.status(200).json(sharingParticipants);
+  }
+);
+
+const sharingRelationParser = z.object({
+  newParticipantSites: z.array(z.number()),
+});
+
+participantsRouter.post(
+  '/:participantId/sharingPermission/add',
+  hasParticipantAccess,
+  async (req: ParticipantRequest, res: Response) => {
+    const { participant } = req;
+    if (!participant?.siteId) {
+      return res.status(400).send('Site id is not set');
+    }
+    const { newParticipantSites } = sharingRelationParser.parse(req.body);
+    const sharingParticipants = await addSharingParticipants(
+      participant.siteId,
+      newParticipantSites
+    );
+
+    const currentUser = await findUserByEmail(req.auth?.payload?.email as string);
```

Comment:
With an audit trail, always insert the audit entry in a way that if the audit insert fails, the operation doesn't happen. So await the insert first, and then submit the update to the admin service.

It's probably a good idea to then update the audit entry afterwards to indicate whether the update succeeded or failed.